### DISCUSSION TITLE
distance "guessing" and data if beacon is not visible

### DISCUSSION THREAD

**Discussion by ralf-koe** [ESPresense/ESPresense#203] [Support]
Hi all,

I like the idea behind this project very much and see active development here :-).

I am currently testing ESPrence with two tile beacons and two ESP32 i my house. One ESP in the first floor and one in the basement.

One beacon is in the middle of the house, nothing moved around, all fixed, but the distance on both ESP is jumping between 5.20-9.30 and 5.20-7.80, around 5.20-5.40 is realistic. 
How can I optimize this?

If I move around with the second beacon, it is not ever visible from both ESP, but there is no data change visible related to "forget beacon". E.g. set distance to 0 or 99, if the beacon is not longer visible.
How can I see, if the data is old? 

And last question for now, on one level of a flat the idea could really work, but how can I see if I am 5 meters away in the same room or 4 meters above in the first floor?

**Comment by DTTerastar** [comment]
Unfortunately because of the propagation of radio waves I've always found upstairs beacons to have a better rssi so it really doesn't work great for detecting elevation.  Try putting both sensors on the same floor and test it that way.

**Comment by ralf-koe** [comment]
Thank you for the answer.

Additonal EPS32 will be here hopefully on Saturday, then I will place them around the groundfloor.

Any hints or ideas regarding jumping values and not longer visible beacons?

**Comment by DTTerastar** [comment]
RSSI is always going to have some amount of jitter and error:  https://www.unicomglobal.com/files/9014/4200/2530/Introduction_to_basic_reflective_multipath_in_short-path_wireless_systems_5-2-13.pdf

I'm not understanding exactly what you are asking beyond that.  Can you clarify?

**Comment by ralf-koe** [comment]
>  some amount of jitter and error

fully clear.

Update: adjusting adblocker and I can upload pictures :-).

![picture 001](https://user-images.githubusercontent.com/28449232/151412046-cd5eaa9e-fbf9-4d73-8683-28add81cbb5d.png)

My test is fully fixed, ESPs and beacon is not moving and the distance reported by espresence is jumping in the last 24 hours between frst ESP 5.20-11.20 and second ESP 4.50-6.50, but both should be more around 5.20-5.40.

Is that normal and the way to optimize it is the use of appdeamon?



Second topic, with another beacon, I get the MQTT messages as long the beacon is reachable by any ESP, but if the beacon is not longer reachble, there is no information about that. The values are not longer updated. Of course, I can use the "update time", but is could be much easier to update an unrealistc value for distance like 0 or 99.

sz last see the beacon at 18:02, but there is no clear hint that the beacon is not reachable.
![picture 002](https://user-images.githubusercontent.com/28449232/151412077-671c8c07-7616-4571-a577-6b83afa1e6ad.png)


I don't use HA, I use a different homeautomation software.

**Comment by DTTerastar** [comment]
Rssi @ 5.5m will have a pretty large error +- 1m is reasonable.  The error decreases with promixity.

Have you tried changing the settings on the ESPresence?  Set maxDistance to 0 so we don't filter out beacons > maxDistance?

**Comment by ralf-koe** [comment]
maxDistance is not set, so I expect the default value of 16.00 is used

Update1: change to 0 and will verify

Update2: There is a difference, the red line is a real ESP32 and values are more or less good, the green line is an m5stack-atom. Possibly that is the problem, I will replace it on the weekend.
